packages/@glimmer/syntax/lib/get-template-locals.ts

```diff
@@ -26,6 +35,10 @@ function tokensFromType(node: ASTv1.Node, scopedTokens: string[]): string | void
       return;
     }
 
+    if (!options.includeHtmlElements && tag.indexOf('.') === -1 && tag.toLowerCase() === tag) {
```

Comment:
Just to confirm, block param cases would have returned already right? (If not, this check isn't sufficient)...

Can you add a test for:

```hbs
{{#let (component @thing) as |button|}}
  <button></button>
{{/let}}
```

Reply:
Yes, this additional check is only for top level items